### PR TITLE
per cell lejepa implementation

### PR DESCRIPTION
**Pull request by shmh40**
## Description

Draft implementation of per-cell LeJEPA with SIGReg loss. LeJEPALoss should probably just go under the LatentLoss class, to do. Also needs checking. 

We have a "SelfTeacher" since we don't do any stop grad/EMA etc. with LeJEPA.

## Issue Number

Closes ???

Is this PR a draft? Mark it as draft.

## Checklist before asking for review

-   [ ] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

**Comment by shmh40** [comment]
image

File: src/weathergen/model/model.py [hunk 1]
```diff
@@ -557,10 +557,11 @@ def create(self) -> "Model":
         self.latent_heads = nn.ModuleDict()
         self.latent_pre_norm = nn.LayerNorm(cf.ae_global_dim_embed)
 
+        ssl_loss_types = ("LossLatentSSLStudentTeacher", "LossLeJEPA")
```
Comment (clessig) [inline src/weathergen/model/model.py:560]: I would prefer if we come up with a consistent naming, e.g. `SSL_` for the ssl losses, so that we do not need to maintain an explicit list here.

File: src/weathergen/train/loss_modules/loss_module_ssl.py
Comment (clessig) [inline src/weathergen/train/loss_modules/loss_module_ssl.py:303]: Why do we need this here? This should be apparent from the masking--forecasting has no mask on the target.

File: src/weathergen/train/loss_modules/loss_module_ssl.py [hunk 2]
```diff
@@ -394,3 +394,134 @@ def get_loss_function_ssl(name):
...
+        num_projections: Number of random unit-vector projection directions.
+        knots: Number of frequency evaluation points in [0, 3].
+
+    Returns:
+        Scalar loss.
+    """
+    z = z.float()  # float32 for cos/sin precision
+    n, d = z.shape
+
+    # Trapezoidal quadrature weights with Gaussian window on [0, 3]
```
Comment (clessig) [inline src/weathergen/train/loss_modules/loss_module_ssl.py:499]: The quadrature should go to a separate function.
